src/compositions/movies.ts

```diff
@@ -0,0 +1,23 @@
+import { reactive, toRefs } from 'vue';
+import { get } from '../plugins/fetch';
+import { RADARR_MOVIES } from '../plugins/fetch/routes/radarr';
+import type { Movie } from '../types/radarr';
+
+const store = reactive({
+    movies: [] as Movie[],
+    loading: false,
+});
+
+const fetchMovies = async (): Promise<void> => {
+    store.loading = true;
+    store.movies = await get<Movie[]>(RADARR_MOVIES);
+    store.loading = false;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/explicit-module-boundary-types
+export const useMovies = () => {
```

Comment:
I don't get how to type this efficiently without redifining all the store objects again. useMovies is correctly typed in VSCode already...

Reply:
Neither `store` nor `fetchMovies` is explicitly typed though, all types are probably inferred, and I can't help but think TypeScript is having a hard time inferring types in `useMovies`. You could try explicitly typing `store` and `fetchMovies`.